### PR TITLE
adding tests to github actions

### PR DESCRIPTION
**Pull request by muskansethi1**
WIP

- [x] Enable tests in github actions
- [x] Fix tests (changes due to malware scanning service)
- [ ] See why database connection is not working in the PR?

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        node: [20, 18]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node }}
+
+      - run: npm i
+      #replicating calesi folder structure
```
Comment (sjvans) [inline .github/workflows/test.yml:31]: why is this necessary? shouldn't it just be `npm i && npm t`?

Comment (muskansethi1) [reply .github/workflows/test.yml:31]: I am cloning incidents-app here, hence replicated the calesi setup so that the tests can run both here and in local in the same setup
